src/styles/spacing.scss

```diff
@@ -1,45 +1,18 @@
-$sizeUnit: rem;
-$marginKey: 'm';
-$paddingKey: 'p';
-$separator: '-';
-$sizes: (
-  ('none', 0),
-  ('xxs', 0.125),
-  ('xs', 0.25),
-  ('sm', 0.5),
-  ('md', 1),
-  ('lg', 2),
-  ('xl', 4),
-  ('xxl', 8),
-);
-$positions: (
-  ('t', 'top'),
-  ('r', 'right'),
-  ('b', 'bottom'),
-  ('l', 'left')
-);
-
-@function sizeValue($key, $value) {
-  @return if($key == 'none', 0, $value + $sizeUnit);
-}
-
-@each $size in $sizes {
-  $sizeKey: nth($size, 1);
-  $sizeValue: nth($size, 2);
-  .#{$marginKey}#{$separator}#{$sizeKey} {
-    margin: sizeValue($sizeKey, $sizeValue);
-  }
-  .#{$paddingKey}#{$separator}#{$sizeKey} {
-    padding: sizeValue($sizeKey, $sizeValue);
-  }
-  @each $position in $positions {
-    $posKey: nth($position, 1);
-    $posValue: nth($position, 2);
-    .#{$marginKey}#{$separator}#{$posKey}#{$separator}#{$sizeKey} {
-        margin-#{$posValue}: sizeValue($sizeKey, $sizeValue);
-    }
-    .#{$paddingKey}#{$separator}#{$posKey}#{$separator}#{$sizeKey} {
-        padding-#{$posValue}: sizeValue($sizeKey, $sizeValue);
+@mixin generateSpacingHelpers($spacingTypes: (margin, padding), $spacingIncrements: 8, $incrementSize: 0.5rem, $spacingSides: (top, right, bottom, left)) {
```

Comment:
@datchley This is the update of the spacing file that can be pulled over to the boilerplate.